README.md

```diff
@@ -1,26 +1,266 @@
 
-Structure:
-- code
-  + functions: we will make a subfolder categories to group functions by generic grouping names, e.g. categories: 'loaders', 'gettersAndSetters', 'plotters', etc...
-  + toolboxes: turtle_son, PAC-master, etc...
-- testDataSets: will be selected from existing datasets or created explicitly for generic testing of function
-- outputFigs: will contain the output of function testing to the specified test dataset
+## Installation Instructions:
+- Compatibility - Mac or PC. We rely on a toolbox to open .json files which does not work on Linux. Requires **Matlab R2019a or prior**. The toolbox we rely on to open .json files is not compatible with Matlab R2019b
+- Clone this repository and add to Matlab path. 
 
-Functions: this list contains the functions that have been tested in brach and pushed to master (brief description of function input output next to each function name)
-- function1: this function is doing X and requires an input argument/s Y1,Y2,Yn and outputs Z
-- ...
+## What is the RC+S native data format?
+The Medtronic API saves data into a session directory. There are 11 .json files which are created for each session, which contain both meta-data and numerical data. These files can contain data from streaming up to 30 hours if streaming time domain data or even longer if streaming power domain data; the limit to this duration is dictated by the max INS battery life. 
 
-In Dev Branch:
-- deserializeJSON: Used to read data from RawDataTD.json into Matlab
-    - fixMalformedJson
-- unravelData: Used in the processing pipeline for converting data in RawDataTD.json to Matlab table format. Takes output from deserializeJSON and converts to Matlab table, without remove any packets  
+There are multiple challenges associated with these .json file and analyzing them: Interpreting metadata within and across the files, handling invalid / missing / misordered packets, creating a timestamp value for each sample, aligning timestamps (and samples) across data streams, and parsing the data streams when there was a change in recording or stimulation parameters. See below for the current approach for how to tackle these challenges.
+
+## Data parsing overview
+
+To facilitate most standard analyses of time-series data, we would optimally like the data formatted in a matrix with samples in rows, data features in columns, and a timestamp assigned to each row. The difference in time between the rows is either 1/Fs or 1/Fs\*x, where x is any whole number multiple. (In the latter case, the user could fill the missing rows with NaNs, if desired). There are many challenges in transforming RC+S data into such a matrix. Here, we provide an overview of the overall approach. More detailed information on specific steps can be found below.
```

Comment:
How should someone pick x? The formula `1/FS\*x` isn't clear to me.

Reply:
I'm trying to communicate that we want to preserve 1/Fs between samples (whether those samples were 'observed' or not). As a toy example, if Fs = 2, we would have times: 0.5, 1, 1.5, 2, 2.5 if there were no samples missing. If there were samples missing, we might have something like this: 0.5, 1, 1.5, 2, 6.5, 7, 7.5. What we want to avoid is something like this: 0.5, 1, 1.5, 2, 6.1, 6.6, 7.1